CredentialProvider.Microsoft/Util/EnvUtil.cs

```diff
@@ -29,6 +29,7 @@ public static class EnvUtil
         public const string BuildTaskUriPrefixes = "VSS_NUGET_URI_PREFIXES";
         public const string BuildTaskAccessToken = "VSS_NUGET_ACCESSTOKEN";
         public const string BuildTaskExternalEndpoints = "VSS_NUGET_EXTERNAL_FEED_ENDPOINTS";
+        public const string ForceCanShowDialogEnvVar = "ARTIFACTS_CREDPROVIDER_FORCE_CANSHOWDIALOG_TO";
```

Comment:
Add this to the EnvironmentVariableHelp help string that has all the other env vars.

Reply:
and to readme manually :(

Reply:
I low-key hate those lists -- they're full of crappy implementation details, bad legacy decisions, and weird inconsistencies, especially since this is used for more than just nuget now

Reply:
Also maybe add a note there that using this env var is not normally recommended.

Reply:
This is somewhat unrelated to this PR but maybe it would make sense to only have the environment variables in the list that we expect people to actually use. Or at least separate the nuget specific variables form the "universal" ones. We could have the full list somewhere else but keep the help string clean.

Reply:
I don't know if I'll ever have time, but I'd like to split the whole thing up into CredentialProvider.Microsoft.exe (NuGet plugin + back-compat wrapper) and artifacts-credprovider.exe (all other use-cases, with all the variables and options renamed and rearranged to actually make sense)

Reply:
That would make a lot of sense